torch/csrc/jit/backends/backend_debug_handler.h

```diff
@@ -7,75 +8,119 @@ namespace torch {
+ * using atomic variable. However this is only used for compilation so we assume
+ * to absorb some of that penalty. Plus if there is no contention then we should
+ * have less to worry about. B. If repeated compilation is part of a long
+ * running process then we may overflow int64_t. We may detect and fail on this.
+ * Not sure if this is good. Will seek opinions about this.
  *
  *  Now about 2:
- *  There are two usecases for debug-handle-to-inlined-callstack-ptr:
- *  A. During bytecode generation the inlined callstack ptrs, corresponding
- *     to the nodes of the inlined graph being serialized, are stored in this
- *     object and a unique debug handle is returned. This unique debug handle
- *     is stored in mobile_debug info for pytorch lite models. It will be used
- *     for raising exceptions as well as profiling.
- *  B. During backend lowering, each backend's preprocess/compile method can
- *     compile method's graph and serialize those methods. Once the method is
- *     lowered to backend, graph is essentially lost. Without access to graph
- *     it is hard to generate model level debug info via source range. Thus
- *     the debug handles provide a way to map nodes of the graph to the model
- * level debug info. Following diffs will provide a API to generate debug
- * handles given inlined graph. This API will, given an inlined graph of a
- * method, returns a map of node*-to-debug-handles. Backends will serialize
- * these debug handles and use them to raise exceptionm, much like lite
- * interpreter. Underneath the API will utilize BackendDebugHandleManager, to
- * generate map of debug-handles-to-inlined-callstack-ptrs.
+ *  There are two usecases for [debug-handle, {source range,
+ * inlined-callstack-ptr}]: A. During bytecode generation the inlined callstack
+ * ptrs and source range, corresponding to the nodes of the inlined graph being
+ * serialized, are stored in this object and a unique debug handle is returned.
+ * This unique debug handle is stored in mobile_debug info for pytorch lite
+ * models. It will be used for raising exceptions as well as profiling. B.
+ * During backend lowering, each backend's preprocess/compile method can compile
+ * method's graph and serialize those methods. Once the method is lowered to
+ * backend, graph is essentially lost. Without access to graph it is hard to
+ * generate model level debug info. Thus the debug handles provide a way to map
+ * nodes of the graph to the model level debug info. Following diffs will
+ * provide an API to generate debug handles given inlined graph. This API will,
+ * given an inlined graph of a method, return a map of node*-to-debug-handles.
+ * Backends will serialize these debug handles and use them to raise exception,
+ * much like lite interpreter. Underneath the API we will utilize
+ * BackendDebugHandleManager, to generate map of [debug-handles, {source range,
+ * inlined-callstack-ptrs}].
```

Comment:
```suggestion
 *  inlined-callstack-ptr}]: A. During bytecode generation the inlined callstack
 *  ptrs and source range, corresponding to the nodes of the inlined graph being
 *  serialized, are stored in this object and a unique debug handle is returned.
 *  This unique debug handle is stored in mobile_debug info for pytorch lite
 *  models. It will be used for raising exceptions as well as profiling. B.
 *  During backend lowering, each backend's preprocess/compile method can compile
 *  method's graph and serialize those methods. Once the method is lowered to
 *  backend, graph is essentially lost. Without access to graph it is hard to
 *  generate model level debug info. Thus the debug handles provide a way to map
 *  nodes of the graph to the model level debug info. Following diffs will
 *  provide an API to generate debug handles given inlined graph. This API will,
 *  given an inlined graph of a method, return a map of node*-to-debug-handles.
 *  Backends will serialize these debug handles and use them to raise exception,
 *  much like lite interpreter. Underneath the API we will utilize
 *  BackendDebugHandleManager, to generate map of [debug-handles, {source range,
 *  inlined-callstack-ptrs}].
```